### ISSUE TITLE
Using Win2D 0.0.0.14 causes app to fail Cert test on Windows Phone store?

### ISSUE THREAD

**Issue by JohnnyWestlake** [Microsoft/Win2D#46]
Alright, so this is an odd one.

Building with the latest Win2D package with VS 2013 Update 4 (or also VS 2015 CTP 5) is good - the package builds. Run Windows App Cert Kit (3.4) locally, the application passes validation - as you would expect.

Now, submit the app to the Windows Phone Store (in my case, the Windows Phone Store as an update to an app using 0.0.0.13) - and it gets to running through the same Windows Cert Kit version on the store - only for it to consistently fail:

```
FAILED
Supported APIs
Error Found: The supported APIs test detected the following errors:
This API is not supported for this application type - Api=12. Module=d2d1.dll. File=Microsoft.Graphics.Canvas.dll.
Impact if not fixed: Using an API that is not part of the Windows SDK for Windows Phone Store apps violates the Windows Phone Store certification requirements.
```
Googling, it seems someone else has the same problem with D2D api's & the Store Cert Kit online https://social.msdn.microsoft.com/Forums/vstudio/en-US/4d93e0c4-aa30-4754-b275-3e4f322ff624/supported-api-test-fail-d2d1?forum=wingameswithdirectx - in his case the use of D2D1::ComputeFlatteningTolerance. (which in Win2D's case may have been used for Geometry support?). It seems like for some reason, even though whatever API is actually causing the issue does seem to actually be supported by Windows Phone and run fine, the store is having none of it.

Now, I don't know where the right place to post this - as it might just be a problem with the store - but I feel it's worth posting here as it seems to make Windows Phone apps un-submitable at the moment. 

(Alternatively, something has gone horribly wrong with my set up - also likely possible)

**Comment by JohnnyWestlake** [comment]
Too add - commenting out everything to do with ComputeFlatteningTolerence in the Win2D source manages to gets it through the cert fine. (I have no idea what API=12 even means or where it comes from, but it seems the phone store doesn't agree with it).

It's an okay solution for me as I don't use it, but I realise it's not something that can be stuck in the library.

**Comment by shawnhar** [comment]
Thank you so much for this report, and for going to extra mile to figure out exactly which method is causing the problem!

This is absolutely the right place to post it.  I will do two things from here:

1) Work with the D2D and Store teams to understand why this API is getting blocked and unblock it.  I'm not sure what the timeline for pushing such a fix out would be, though  (can the cert process update any time, or is that tied to new SDK or OS versions?) so in the meantime:

2) Find some way to work around the issue in Win2D.  Obviously not being able to publish your app is a problem that we can't afford to ignore ;-)

**Comment by shawnhar** [comment]
This is fixed in Win2D 0.0.14, which we pushed today.

Full disclosure:  I hope it is fixed but haven't actually verified by publishing an app all the way to the store yet :-)   Having learned that it is possible to hit issues like this that can't be tested locally, we plan to go all the way through publishing with Example Gallery soon, but there's a little more cleanup needed before we'll be able to do that.  So for now, I just changed the implementation to avoid calling D2D1::ComputeFlatteningTolerance.   Fingers crossed this will take care of it...

Thanks for the report!
